src/metatrain/utils/llpr.py

```diff
@@ -0,0 +1,426 @@
+from typing import Dict, List, Optional
+
+import metatensor.torch
+import numpy as np
+import torch
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatensor.torch.atomistic import (
+    ModelCapabilities,
+    ModelEvaluationOptions,
+    ModelOutput,
+    System,
+)
+from torch.utils.data import DataLoader
+
+
+class LLPRModel(torch.nn.Module):
```

Comment:
Since all the quantities given are uncertainties and not the rigidities, it may be more reasonable to label this is `LLPRUncertaintyModel`